MQ2DanNet/MQ2DanNet.cpp

```diff
@@ -2571,6 +2571,16 @@ std::string unescape_string(const std::string& input) {
     return std::string(szOut);
 }
 
+// The name sent to be checked against mq2dannet must be all lower case otherwise this will always return false.
+PLUGIN_API bool MQ2DanNetAreTheyConnected(char* szName)
+{
+	auto peers = Node::get().get_peers();
+	std::set<std::string> connectedcharacters;
+	std::transform(peers.cbegin(), peers.cend(), std::inserter(connectedcharacters, connectedcharacters.begin()),
+		[](std::string s) -> std::string {return Node::get().get_short_name(s); });
```

Comment:
This doesn't respect the `full_names` boolean, so if you have people with characters across servers, you can't guarantee uniqueness. Instead of doing this transform, use the `get_name` function on the passed value and search `peers` for that. You don't actually need to transform `peers` at all.

Reply:
I considered doing that, but honestly didn't consider it that important.  I just want to white list characters for mq2posse that are on your network at that moment.

Reply:
If it makes you happy, I can use the full_names.

Reply:
I generally like the idea of following the vision of people who's code I'm playing with (assuming they are still active), it just makes everything easier.